ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -1331,6 +1340,15 @@ public void dropTable(String tableName, boolean ifPurge) throws HiveException {
     dropTable(names[0], names[1], true, true, ifPurge);
   }
 
+  public void dropTable(Table table, boolean ifPurge) throws HiveException {
+    boolean deleteData = !((HiveConf.getBoolVar(conf, ConfVars.HIVE_TXN_NON_BLOCKING_DROP_TABLE)
+        || HiveConf.getBoolVar(conf, ConfVars.HIVE_TXN_LOCKLESS_READS_ENABLED))
+      && AcidUtils.isTransactionalTable(table)
+      && table.getSd().getLocation().matches(PATH_WITH_VERSION_ID));
```

Comment:
Idk if there's any getting around this, but what if the actual table name matches ".*_v\d" ?

Also if a table was created with HIVE_TXN_NON_BLOCKING_DROP_TABLE=false, we can still keep the data right? so we can get rid of : && table.getSd().getLocation().matches(PATH_WITH_VERSION_ID)

Reply:
if table location doesn't contain the versioning - we won't support lockless reads